src/util/amount.rs

```diff
@@ -995,6 +995,184 @@ pub mod serde {
     }
 }
 
+/// Type for dealing with unknown amounts
+#[cfg_attr(feature = "schemars", derive(schemars::JsonSchema))]
+#[derive(Debug, PartialEq)]
+pub enum CoinAmount {
+    /// Sats type
+    Sats(u64),
+    /// Btc type
+    Btc(f64),
+}
+
+/// Derived externally using serde_derive
+#[cfg(feature = "serde")]
+#[doc(hidden)]
+#[allow(non_upper_case_globals, unused_attributes, unused_qualifications)]
+const _: () = {
+    #[allow(rust_2018_idioms, clippy::useless_attribute)]
+    extern crate serde as _serde;
+    impl _serde::Serialize for CoinAmount {
+        fn serialize<__S>(&self, __serializer: __S) -> _serde::export::Result<__S::Ok, __S::Error>
```

Comment:
If this is auto generated by `serde_derive` I'd recommend cleaning it manually since if later needs to be updated one would need to go through the process of regenerating the serde code for the enum instead of modifying the code directly.

Something along the lines of https://github.com/rust-bitcoin/rust-bitcoin/blob/master/src/blockdata/script.rs#L757-L767 could be done

Reply:
Actually to me (maybe incorrectly) it seems simpler to only minimally modify the generated code for that exact reason -- one can just regenerate it and patch as needed.

In any case the example you shared is also simpler, since that is a struct and this is an enum.

Reply:
Hmm yeah I'd prefer not to have these code-generated impls in here :/

Reply:
I'm not sure where or what you want exactly then?

Reply:
IMHO renaming the variables for reading them and formatting the code to fit the rest of the file would be enough.